src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/LimitedConcurrencyLevelTaskScheduler.cs

```diff
@@ -0,0 +1,186 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Data.SqlClient.SNI
+{
+    /// <summary>
+    /// Limited concurrent level task scheduler
+    /// </summary>
+    internal class LimitedConcurrencyLevelTaskScheduler : TaskScheduler
```

Comment:
I don't think you should need this.  If your goal is just to be able to control the max number of tasks that can run concurrently, you can construct a ConcurrentExclusiveTaskScheduler with that max number, and then schedule to its ConcurrentScheduler.